### ISSUE TITLE
Is there any example code about copy variable parameters between two Program?

### ISSUE THREAD

**Issue by TheodoreG** [PaddlePaddle/Paddle#14443]
Hey guys here I have an question, how to copy parameters between Programs? 
Suppose that I have defined two Programs and they have same structure in the last few layers,  for example they both have two FC layers. While I print the variables I got:

For program A:
[u'fc1.b_0', u'fc1.tmp_0', u'fc1.tmp_1', u'fc1.w_0', u'fc2.b_0', u'fc2.tmp_0', u'fc2.tmp_1', u'fc2.tmp_2', u'fc2.w_0', u'feature', u'label', u'mean_0.tmp_0', u'top_k_0.tmp_0', u'top_k_0.tmp_1', u'top_k_1.tmp_0', u'top_k_1.tmp_1']

For program B:
[ ..........(many other layers)......,  u'fc1.b_1', u'fc1.tmp_2', u'fc1.tmp_3', u'fc1.w_1', u'fc2.b_1', u'fc2.tmp_3', u'fc2.tmp_4', u'fc2.tmp_5', u'fc2.w_1', u'feature.tmp_0', u'feature.tmp_1', u'image', u'label', u'mean_1.tmp_0', u'pool2d_0.tmp_0', u'pool2d_1.tmp_0', u'top_k_2.tmp_0', u'top_k_2.tmp_1', u'top_k_3.tmp_0', u'top_k_3.tmp_1']

Now I know exactly those the FC layers in two Programs share same structure and just have different variable name. How can I copy the parameter of 'fc1.b_0' of Program A into 'fc1.b_1' of Program B and so on?

**Comment by Xreki** [comment]
To share parameters between two programs, the simplest way is sharing the parameters' name. You can use `param_attr` to specify the parameters' name of those operators. For example,

```python
embed_first = fluid.layers.embedding(
            input=words[0],
            size=[dict_size, EMBED_SIZE],
            dtype='float32',
            is_sparse=IS_SPARSE,
            param_attr='shared_w')
```

With the same names, and running in the same scope, the parameters will be automated shared.

**Comment by TheodoreG** [comment]
Thank you, I still have two questions.
1. I used following function to check if the parameters in two programs are the same.
```
    mid = fluid.layers.fc(input=self.pool,
                              size=1024,
                              act='relu',
                              name='fc1',
                              param_attr='fc1_shared_w',
                              bias_attr='fc1_shared_b')
     mid = fluid.layers.dropout(mid,
                                   dropout_prob=self.model_conf["NET.dropout_rate"],
                                   name='dropout')
     self.out = fluid.layers.fc(input=mid,
                                   size=self.model_conf["class_num"],
                                   act='softmax',
                                   name='fc2',
                                   param_attr='fc2_shared',
                                   bias_attr='fc2_shared_b')
```
`fc1_shared_w_test = fluid.io.get_parameter_value_by_name(u'fc1_shared_w', exe, test_program)`
`fc1_shared_w_infer = fluid.io.get_parameter_value_by_name(u'fc1_shared_w', exe, infer_program)`
But I got some errors:

  fc1_shared_w_test = fluid.io.get_parameter_value_by_name(u'fc1_shared_w', exe, test_program)
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/io.py", line 878, in get_parameter_value_by_name
    return get_parameter_value(var, executor)
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/io.py", line 845, in get_parameter_value
    return executor.run(get_program, feed={}, fetch_list=[new_var])[0]
  File "/usr/local/lib/python2.7/dist-packages/paddle/fluid/executor.py", line 470, in run
    self.executor.run(program.desc, scope, 0, True, True)
EnforceNotMet: holder_ should not be null
Tensor holds no memory. Call Tensor::mutable_data first. at [/paddle/paddle/fluid/framework/tensor.cc:22]

2. How do I make sure that two Programs are in the same scope?

**Comment by Xreki** [comment]
1. Did you run the program? Memory will not be allocated until you run the program.

2. A default scope will be used if no user-defined scope is specified. If you run under `with fluid.scope_guard(some_scope)`, then `some_scope` is the default scope. Just ensure `exe.run` is called under the same scope.

**Comment by lucywsq** [comment]
Excuse me, if your problem has been solved，we would  close it.  If you still need to follow up on this question after closing, please feel free to reopen it. Thank you so much for your support of PaddlePaddle Group!

**Comment by TheodoreG** [comment]
HI, I have got to this problem again . I am using fluid parallel training with "rpc" mode and "all_reduce" strategy. I have two fc layers to train, after that I need to attach this two layers to a pretrained resnet50 model. Suppose these two layers are named as "fc1_shared_w" and "fc2_shared_w". When I create a infer model it shows:
![image](https://user-images.githubusercontent.com/13977118/57674372-b99ccb80-7651-11e9-90f2-1d00e9707eaa.png)

In non-distributed training this wrong would not happen. I found that it is due to two initialization operations. Here are those two ops:

![image](https://user-images.githubusercontent.com/13977118/57674533-19937200-7652-11e9-8649-194afe835998.png)


![image](https://user-images.githubusercontent.com/13977118/57674617-519ab500-7652-11e9-8612-40202a816bc0.png)

It seems that in distributed mode those on parameter has two initialization op and cause the problem. My paddle version is 1.1.0. How to solve it ?
